### PR TITLE
Add basic code coverage/API tests for RFSA

### PR DESCRIPTION
**Pull request by astarche**
### What does this Pull Request accomplish?

Add tests that exercise the different codegen features in the RFSA service to make sure we get reasonable results.

### Why should this Pull Request be merged?

Improve validation of RFSA service.

### What testing has been done?

Ran and passed included tests. Except `GetRelayName_ReturnsRelayName`, which currently fails due to [AB#1639825](https://ni.visualstudio.com/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/1639825).

### CODE REVIEW

File: .github/workflows/build_cmake.yml [hunk 1]
```diff
@@ -26,16 +26,12 @@ jobs:
...
-            # On msvc, the -j flag does not help build times and causes intermittent
-            # file permission errors.
-            cmake_number_of_jobs: ""
           }
         - {
             name: "Ubuntu Build", artifact: "Linux.tar.xz",
             os: ubuntu-18.04,
             build_type: "Release", cc: "gcc-9", cxx: "g++-9",
             cmake_platform: "",
-            cmake_number_of_jobs: "-j 2",
```
Comment (astarche) [inline .github/workflows/build_cmake.yml:38]: Same with nilrt, ubuntu went over the edge.

The "good news" is that it's increasingly clear that the problems are with core_server and device_server including everything. We can fix that, and I have a plan for it. So, as long as we can implement that and keep future services DAQ-size-or-smaller, we can turn this back on.

**Review by gregstoll — APPROVED** [review]

File: .github/workflows/build_cmake.yml (hunk 1 above)
Comment (gregstoll) [reply .github/workflows/build_cmake.yml:38]: I'm pretty sure this is causing the build on #370 to fail as well, so I'll wait for this to go in first.

Comment (astarche) [reply .github/workflows/build_cmake.yml:38]: I'll just separate this out and push it through

File: source/tests/system/nirfsa_driver_api_tests.cpp [hunk 2]
```diff
@@ -21,38 +40,340 @@ class NiRFSADriverApiTests : public Test {
...
+  EXPECT_SUCCESS(session, response);
+  EXPECT_THAT(response.frequencies(), Each(Not(Eq(0))));
+  EXPECT_THAT(response.phase_response(), Each(Eq(0)));
+  EXPECT_THAT(response.magnitude_response(), Each(Eq(0)));
+  EXPECT_EQ(EXPECTED_SIZE, response.frequencies().size());
+  EXPECT_EQ(EXPECTED_SIZE, response.phase_response().size());
+  EXPECT_EQ(EXPECTED_SIZE, response.magnitude_response().size());
+}
+
+// TODO: AB#1639825. This currently fails because of an off-by-one error.
```
Comment (reckenro) [inline source/tests/system/nirfsa_driver_api_tests.cpp:161]: Does that this mean that this test should be skipped or commented out for now then?

**Review by reckenro — APPROVED** [review]

File: source/tests/system/nirfsa_driver_api_tests.cpp (hunk 2 above)
Comment (reckenro) [reply source/tests/system/nirfsa_driver_api_tests.cpp:161]: Oh nevermind, these are system tests so they're run manually.

Comment (astarche) [reply source/tests/system/nirfsa_driver_api_tests.cpp:161]: It looks like the gtest standard is a DISABLED_ prefix. I can do that.

**Comment by astarche** [comment]
Bypassing @SparkingSpork . More j2. Need to fix the build.

**Pull request merged**